stable/openstack-base/README.md

```diff
@@ -1,333 +1,238 @@
-	done
+We'll also need an 'internal' network for the admin user which instances are
+actually connected to:
 
-You only need to perform this step once.
+    openstack network create internal
 
-## Create an instance
+    openstack subnet create --network internal \
+        --subnet-range 198.51.100.0/24 \
+        --dns-nameserver 8.8.8.8 \
+        internal_subnet
 
-To create a Bionic instance called 'bionic-1':
+    openstack router create provider-router
 
-    openstack server create --image bionic --flavor m1.tiny --key-name mykey \
-        --nic net-id=$(openstack network list | grep internal | awk '{ print $2 }') \
-        bionic-1
+    openstack router set --external-gateway ext_net provider-router
 
-## Attach a volume
+    openstack router add subnet provider-router internal_subnet
 
-This step is optional.
+Neutron provides a wide range of configuration options; see the [OpenStack Neutron][] documentation for more details.
 
-To create a 10GiB volume in Cinder and attach it to the new instance:
+### Configuring a flavor
 
-    openstack volume create --size=10 <volume-name>
-    openstack server add volume bionic-1 <volume-name>
+Starting with the OpenStack Newton release, default flavors are no longer created at install time. You therefore need to create at least one machine type before you can boot an instance:
 
-The volume becomes immediately available to the instance. It will however need
-to be formatted and mounted before usage.
+    openstack flavor create --ram 2048 --disk 20 --ephemeral 20 m1.small
 
-## Assign a floating IP address
+### Booting an instance
 
-To request and assign a floating IP address:
+First generate a SSH keypair so that you can access your instances once you've booted them:
 
-    FLOATING_IP=$(openstack floating ip create -f value -c floating_ip_address ext_net)
-    openstack server add floating ip bionic-1 $FLOATING_IP
+    mkdir -p ~/.ssh
+    touch ~/.ssh/id_rsa_cloud
+    chmod 600 ~/.ssh/id_rsa_cloud
+    openstack keypair create mykey > ~/.ssh/id_rsa_cloud
 
-## Log in to an instance
+**Note:** you can also upload an existing public key to the cloud rather than generating a new one:
 
-To log in to the new instance:
+    openstack keypair create --public-key ~/.ssh/id_rsa.pub mykey
 
-    ssh -i ~/.ssh/id_mykey ubuntu@$FLOATING_IP
+You can now boot an instance on your cloud:
 
-## Access the cloud dashboard
+    openstack server create --image focal --flavor m1.small --key-name mykey \
+        --network internal focal-test
 
-The cloud dashboard (Horizon) can be accessed by going to:
+### Attaching a volume
 
-`http://<openstack-dashboard-ip>/horizon`
+First, create a 10G volume in cinder:
 
-The IP address is taken from the output to:
+    openstack volume create --size=10 <name-of-volume>
 
-    juju status openstack-dashboard
+then attach it to the instance we just booted:
 
-Print the credentials in this way:
```

Comment:
Now to get the credentials there is a [juju action](https://docs.openstack.org/charm-guide/latest/openstack-on-lxd.html#openstack-dashboard). It would be useful to mention it here as this is typically the kind of things on which a new user would waste 1-2 hours (speaking from experience ;) )

Reply:
Added a section on how to get to and find credentials for dashboard.